src/hera/workflows/retry_strategy.py

```diff
@@ -36,7 +36,7 @@ class RetryStrategy(_BaseModel):
     affinity: Optional[RetryAffinity] = None
     backoff: Optional[Backoff] = None
     expression: Optional[str] = None
-    limit: Optional[Union[int, str]] = None
+    limit: Optional[IntOrString] = None
```

Comment:
Thanks for submitting the PR @lakhanjindam! 🚀 

There might be something strange happening here.. there's a `validator` for this field that is supposed to convert the `limit` from `str | int` to `IntOrString`

Reply:
problems seems to be happening with this line https://github.com/argoproj-labs/hera/blob/main/src/hera/workflows/retry_strategy.py#L55 
since i made change where `limit` is mapped to `IntOrString` type so it goes into if block and returns from there.

Reply:
what we can do is instead of that line we can add something like 
```python
def _convert_limit(cls, v):
  if v is None or isinstance(v, IntOrString):
    return v

  v = cast(IntOrStr, v)
  return v.value
```

Reply:
Ah, yes, you are right... I think, though, that we should enrich this type so that users can pass in 1 of 3 types:
```suggestion
    limit: Optional[Union[int, str, IntOrString]] = None
```

Also, you pointed out the `__root__` usage issue. If we change line 55 from 
```python
return IntOrString(__root__=str(v)) # int or str
```
to
```python
return str(v)  # int or str
```

everything should work.

Reply:
yeah its working that way, if we just return **stringified** version of value but what if the value passed is `bool` then it will just convert that to string and no err will be raised ?

Reply:
yeah this will work. But what if the value passed is `bool` for var `limit`, in that case it wont raise any err since it will be converting it to `str` type